components/mosart/src/riverroute/MOSART_BGC_type.F90

```diff
@@ -0,0 +1,225 @@
+
+module MOSART_BGC_type
+
+!-----------------------------------------------------------------------
+!BOP
+!
+! !MODULE: MOSART_BGC_type
+!
+! !DESCRIPTION:
+! Module containing data structure for riverine biogeochemistry
+!
+!! !REVISION HISTORY:
+! Hongyi Li: Created 03/2015
+!USES:
+  use mct_mod
+  use RunoffMod     , only : Tctl, TUnit, rtmCTL
+  use RtmSpmd       , only : masterproc
+  use RtmVar        , only : iulog
+  use RtmIO         , only : pio_subsystem, ncd_pio_openfile, ncd_pio_closefile
+  use rof_cpl_indices, only : nt_rtm
+  use shr_kind_mod  , only : r8 => shr_kind_r8, SHR_KIND_CL
+  use shr_const_mod , only : denh2o => SHR_CONST_RHOFW, denice => SHR_CONST_RHOICE, grav => SHR_CONST_G, SHR_CONST_REARTH, SHR_CONST_PI
+  use shr_sys_mod   , only : shr_sys_flush, shr_sys_abort
+  use MOSART_RES_type  , only : Tres_ctl, Tres_para, Tres
+  use netcdf
+  use pio
+
+
+! !PUBLIC TYPES:
+  implicit none
+  real(r8), parameter :: TINYVALUE_s = 1.0e-14_r8  ! double precision variable has a significance of about 16 decimal digits
+
+  private
+
+! control information for subbasin-based representation
+
+  ! sediment parameters
+  type Tparameter_sedi
+     character(len=200) :: paraFile             ! the path of the parameter files
+     character(len=100) :: inputPath            ! the path of the input files
+
+     real(r8), pointer :: d50(:)                ! median bed-material sediment particle size [m]
+
+     real(r8), pointer :: clay_content(:)       ! clay content in top soil (-)
+     real(r8), pointer :: silt_content(:)       ! silt content in top soil (-)
+     real(r8), pointer :: rock_frac(:)          ! rock fragment fraction in top soil (-)
+     real(r8), pointer :: crop_frac(:)          ! cropland fraction in landcover (-)
+     real(r8), pointer :: annual_Qsur(:)        ! annual runoff (mm), for soil erosion estimation with an empirical formula
+     real(r8), pointer :: annual_Qtot(:)        ! annual runoff (mm), for soil erosion estimation with an empirical formula
+     real(r8), pointer :: peak_Qtot(:)          ! peak daily runoff (mm), for soil erosion estimation with an empirical formula
+     real(r8), pointer :: peak_Qsur(:)          ! peak daily runoff (mm), for soil erosion estimation with an empirical formula
+     real(r8), pointer :: u_f(:)                ! formative shear bed shear velocity [m/s]
+     logical           :: first_call = .true.
+  end type Tparameter_sedi 
+
+    ! sediment storage and flux variables
+    ! note most of the storage/flux items are included in runoff_flow structure in the RunoffMod.F90
+    ! here we just add some unique items for sediment such as shear stress etc.
+    !public :: TstatusFlux_sedi
+    type TstatusFlux_sedi
+
+        ! hillsloope
+        real(r8), pointer :: Tau_h(:)       ! shear stress for hillslope erosion, [N/m2]
```

Comment:
See my comment above for these two variables.

Reply:
Removed